scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py

```diff
@@ -0,0 +1,258 @@
+
+
+def _do_single_plot(ax, workspace, index, errors, plot_kwargs):
+    plot_fn = ax.errorbar if errors else ax.plot
+    plot_kwargs['wkspIndex'] = index
+    plot_fn(workspace, **plot_kwargs)
+
+
+class PlottingCanvasView(QtWidgets.QWidget, PlottingCanvasViewInterface):
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        # create the figure
+        self.fig = Figure()
+        self.fig.canvas = FigureCanvas(self.fig)
+        self.toolBar = PlotToolbar(self.fig.canvas, self)
+
+        # Create a set of Mantid axis for the figure
+        self.fig, axes = get_plot_fig(overplot=False, ax_properties=None, axes_num=1,
+                                      fig=self.fig)
+        self._number_of_axes = 1
+
+        layout = QtWidgets.QVBoxLayout()
+        layout.addWidget(self.toolBar)
+        layout.addWidget(self.fig.canvas)
+        self.setLayout(layout)
+
+        self._plot_information_list = []  # type : List[PlotInformation}
+
+    @property
+    def plotted_workspace_information(self):
+        return self._plot_information_list
+
+    @property
+    def plotted_workspaces_and_indices(self):
+        plotted_workspaces = []
+        plotted_indices = []
+        for plot_info in self._plot_information_list:
+            plotted_workspaces.append(plot_info.workspace_name)
+            plotted_indices.append(plot_info.index)
+
+        return plotted_workspaces, plotted_indices
+
+    @property
+    def num_plotted_workspaces(self):
+        return len(self._plot_information_list)
+
+    @property
+    def number_of_axes(self):
+        return self._number_of_axes
+
+    def add_widget(self, widget):
+        self.layout().addWidget(widget)
+
+    def create_new_plot_canvas(self, num_axes):
+        """Creates a new blank plotting canvas"""
+        self._plot_information_list = []
+        self._number_of_axes = num_axes
+        self.fig.clf()
+        self.fig, axes = get_plot_fig(overplot=False, ax_properties=None, axes_num=num_axes,
+                                      fig=self.fig)
+        self.fig.tight_layout()
+        self.fig.canvas.draw()
+
+    def clear_all_workspaces_from_plot(self):
+        """Clears all workspaces from the plot"""
+        for ax in self.fig.axes:
+            ax.cla()
+            ax.tracked_workspaces.clear()
+            ax.set_prop_cycle(None)
+        self._plot_information_list = []
+
+    def add_workspaces_to_plot(self, workspace_plot_info_list: List[WorkspacePlotInformation]):
+        """Add a list of workspaces to the plot - The workspaces are contained in a list PlotInformation
+        The PlotInformation contains the workspace name, workspace index and target axis."""
+        for workspace_plot_info in workspace_plot_info_list:
+            workspace_name = workspace_plot_info.workspace_name
+            try:
+                workspace = AnalysisDataService.Instance().retrieve(workspace_name)
+            except RuntimeError:
+                continue
```

Comment:
could this be a return?

Reply:
Not in this case, as its plotting a list of workspaces, this just means this workspace was invalid, but the others may not be. So the for loop needs to continue.